vortex-expr/src/binary.rs

```diff
@@ -76,3 +76,183 @@ impl PartialEq<dyn Any> for BinaryExpr {
             .unwrap_or(false)
     }
 }
+
+/// Create a new `BinaryExpr` using the `Eq` operator.
+///
+/// ## Example usage
+///
+/// ```
+/// use vortex_array::array::{BoolArray, PrimitiveArray };
+/// use vortex_array::{IntoArrayData, IntoArrayVariant};
+/// use vortex_array::validity::Validity;
+/// use vortex_buffer::buffer;
+/// use vortex_expr::{eq, ident, lit};
+///
+/// let xs = PrimitiveArray::new(buffer![1i32, 2i32, 3i32], Validity::NonNullable).into_array();
+/// let result = eq(ident(), lit(3)).evaluate(&xs).unwrap();
+///
+/// assert_eq!(
+///     result.into_bool().unwrap().boolean_buffer(),
+///     BoolArray::from_iter(vec![false, false, true]).boolean_buffer(),
```

Comment:
One thing I noticed while writing these doctests is that we don't have a nicer generic way of building a BooleanBuffer. You need to build a full BoolArray and then access its arrow booleanbuffer.

I added a macro so you could do something like `bools![true, false, true]` but that still forces you to bring in an arrow_buffer dep. We should probably use and return our own booleanbuffer type to make this more ergonomic

Reply:
Yeah, we need to make our own BooleanBuffer type... relatedly, https://github.com/spiraldb/vortex/issues/1774

Reply:
You can construct `BooleanBuffer` directly from `Iter<bool>`

Reply:
Yea but then you need to add a dep on arrow_buffer where there wasn't previously, and sadly rust doesn't let you enable a dep _only_ for doctests

Reply:
oh actually I'm wrong, apparently doc tests use dev-dependencies